rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -0,0 +1,700 @@
+        };
+    }
+
+    /**
+     * Validate response and response type.
+     * @param response Client response.
+     * @param responseType Response type.
+     * @throws RestClientException Thrown in case validation fails.
+     */
+    private void validateResponse(ClientResponse response, ParameterizedTypeReference<?> responseType) throws RestClientException {
+        if (response == null) {
+            throw new RestClientException("Missing response");
+        }
+        if (responseType == null) {
+            throw new RestClientException("Missing response type");
+        }
+        if (response.statusCode().isError()) {
+            ResponseEntity<String> rawResponseEntity = response.toEntity(String.class).block();
+            String rawResponse = null;
+            HttpHeaders rawResponseHeaders = null;
+            if (rawResponseEntity != null) {
+                rawResponse = rawResponseEntity.getBody();
+                rawResponseHeaders = rawResponseEntity.getHeaders();
+            }
+            // Try to parse ErrorResponse in case expected response type is ObjectResponse
+            Class<?> clazz = TypeFactory.rawClass(responseType.getType());
+            if (clazz.isAssignableFrom(ObjectResponse.class)) {
+                try {
+                    // Use an ObjectMapper to deserialize the error response
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    ErrorResponse errorResponse = objectMapper.readValue(rawResponse, ErrorResponse.class);
+                    if (errorResponse != null) {
+                        throw new RestClientException("HTTP error occurred: " + response.statusCode(), response.statusCode(), rawResponse, rawResponseHeaders, errorResponse);
+                    }
+                } catch (IOException ex) {
+                    // Exception is handled silently, ErrorResponse is not available, use a regular error with raw response
+                }
+            }
+            throw new RestClientException("HTTP error occurred: " + response.statusCode(), response.statusCode(), rawResponse, rawResponseHeaders);
+        }
+    }
+
+    /**
+     * Build URI based on configuration of base URL and path.
+     * @param uriSpec Request headers URI specification.
+     * @param path URI path.
+     * @return Request header specification.
+     */
+    private WebClient.RequestHeadersSpec<?> buildUri(WebClient.RequestHeadersUriSpec<?> uriSpec, String path) {
+        if (config.getBaseUrl() == null) {
+            return uriSpec.uri(path);
+        } else {
+            return uriSpec.uri(uriBuilder -> uriBuilder.path(path).build());
+        }
+    }
+
+    /**
+     * Build URI based on configuration of base URL and path.
+     * @param uriSpec Request headers URI specification.
+     * @param path URI path.
+     * @return Request header specification.
+     */
+    private WebClient.RequestBodySpec buildUri(WebClient.RequestBodyUriSpec uriSpec, String path) {
```

Comment:
dtto

I think we should rename this method and maybe improve documentation. This no longer just builds the URI as it used to...

Reply:
Fixed.